Added encrypt key handling for package page (SCRD-7652)

guglo opened the pull request:
1.When load the package page, it will prompt user for encrypt key if
the cloud config is encrypted.
2.The encrypt key is saved in cache, refresh the page, it will prompt user
for encrypt key again if cloud config is encrypted.

Commit:
Added encrypt key handling for package page (SCRD-7652)

1.When load the package page, it will prompt user for encrypt key if
the cloud config is encrypted.
2.The encrypt key is saved in cache, refresh the page, it will prompt user
for encrypt key again if cloud config is encrypted.

guglo commented:
It goes with https://gerrit.suse.provo.cloud//#/c/5673

GarySmith commented on src/localization/bundles/en.json:
```diff
@@ -71,7 +71,7 @@
     "common.encryptkey.message": "Please provide encryption key for running playbooks",
     "common.encryptkey.title": "Provide encryption key",
-    "common.encryptkey.description": "The operation you are about to perform requires the encryption key. Please provide the encryption key.",
+    "common.encryptkey.description": "The operation you are about to perform requires the encryption key. Please provide the encryption key. If provide wrong encryption key, operation will fail. Refresh the page to invoke this dialog again in case wrong key was provided. ",
```
The wording here needs a little fine-tuning. Consider:

The operation you are about to perform runs a playbook that requires the encryption key. If you provide wrong encryption key, the playbook will fail and you will need to refresh your browser window in order to re-enter the encryption key.

guglo replied:
ok, will update. Thanks for the wordings fixes.

guglo commented on src/pages/ArdanaPackages.js:
```diff
@@ -29,13 +31,42 @@ class ArdanaPackages extends Component {
+    if(isEncrypted) {
+      // use the window to save it in cache so it will
+      // not prompt every time when access this page
+      if(isEmpty(window.encryptKey)) {
```
@GarySmith  ok will give it a try.

guglo commented on src/pages/ArdanaPackages.js:
```diff
@@ -29,13 +31,42 @@ class ArdanaPackages extends Component {
+      if(isEmpty(window.encryptKey)) {
+        this.setState({showEncryptKeyModal: true});
+      }
+      else {
```
ok will update

Commit:
Addressed review comments:

Switched to use global var in package page instead of window var
Updated the message

Commit:
Minor update

guglo commented:
To test is_encrypted API call in development env, from ardana-service dir
mkdir data/ardana-ansible/group_vars
echo '$ANSIBLE_VAULT;1.1;AES256' >data/ardana-ansible/group_vars/all --> to test true
echo '1.1;AES256' > data/ardana-ansible/group_vars/all --> test false
Not sure where to put this instruction permanently though.

dotNomad commented on src/localization/bundles/en.json:
```diff
@@ -71,7 +71,7 @@
     "common.encryptkey.message": "Please provide encryption key for running playbooks",
     "common.encryptkey.title": "Provide encryption key",
-    "common.encryptkey.description": "The operation you are about to perform requires the encryption key. Please provide the encryption key.",
+    "common.encryptkey.description": "The operation you are about to perform runs a playbook that requires the encryption key. If you provide the wrong encryption key, the playbook will fail and you will need to refresh your browser window in order to re-enter the encryption key.",
```
I think this can be written a bit more generally that doesn't put as much blame on the user. Perhaps:

> The operation you are about to perform runs a playbook that requires the encryption key. If the encryption key provided is incorrect the playbook will fail and the browser window will need to be refreshed in order to re-enter the encryption key.

guglo replied:
@dotNomad  will rewording it. thanks.

dotNomad reviewed (approved):
A quick rewording suggestion. Looks good though, and works as intended. :+1:

Commit:
Merge branch 'master' into pkgenkey

Commit:
Updated the message

GarySmith commented:
LGTM, but need to be able to re-use this encryptKey throughout the application. Will create a defect for that.
